cmd/livepeer/livepeer.go

```diff
@@ -183,17 +191,17 @@ func main() {
 	} else if transcoder {
 		n.NodeType = core.TranscoderNode
 		glog.Info("***Livepeer is in transcoder mode ***")
-		server.RunTranscoder(n, *orchAddr)
+		server.RunTranscoder(n, orchAddresses[0])
 		return
 	} else {
 		n.NodeType = core.BroadcasterNode
 	}
 
 	if n.NodeType == core.BroadcasterNode {
-		if *orchAddr == "" {
-			glog.Info("No orchestrator specified; transcoding will not happen")
+		if len(orchAddresses) <= 0 {
+			n.OrchestratorSelector = discovery.NewOffchainOrchestratorFromOnchainList(n)
```

Comment:
We probably need a couple more checks around this area:

* Check for offchain mode somewhere (indicated by `n.Eth = nil`). Can either be done right here, or within `NewOffchainOrchestratorFromOnchainList`. I'm inclined to do it within the latter function, and return nil in offchain mode.
* Handle the case where `n.OrchestratorSelector` is nil, and accordingly log "No orchestrator specified; transcoding will not happen". This check should cover the results from either `NewOffchainOrchestrator` or `NewOffchainOrchestratorFromOnchainList`

Reply:
Is there a danger to `n.Eth = client` not being set? We're returning an error when calling `eth.NewClient(ethcommon.HexToAddress(*ethAcctAddr), keystoreDir, backend, ethcommon.HexToAddress(*controllerAddr), EthTxTimeout)`, which should cause the function to return. What do you think?

Reply:
Just checked: it does crash at this point if `-offchain` mode is set on the broadcaster.